### PR TITLE
Adding task to predict location given images

### PR DESCRIPTION
**Pull request by francesco98**

### CODE REVIEW

File: discojs/discojs-node/src/tasks/geotags.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { tf, Task } from '../core'
```
Comment (s314cy) [inline discojs/discojs-node/src/tasks/geotags.ts]: the task should be defined in `discojs-core` only

File: discojs/discojs-web/src/tasks/geotags.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  const model = tf.sequential({
+    layers: [
+      tf.layers.inputLayer({ inputShape: [224, 224, 3] }),
+      tf.layers.rescaling({ scale: 1 / 127.5, offset: -1 }), // Rescaling input between -1 and 1
+      pretrainedModel
+    ]
+  })
+
+  await model.save('indexeddb://saved/geotags/model-v1')
```
Comment (s314cy) [inline discojs/discojs-web/src/tasks/geotags.ts]: this line is great to quickly test your model locally, however a task should only be defined in discojs-core to avoid any confusion

for now, this means moving the geotag task to discojs-core, thus users will need to first train your task before proceeding to validate the model saved in indexeddb (this will be fixed by #497)

File: example_training_data/geotags-labels.csv [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿0,0
```
Comment (s314cy) [inline example_training_data/geotags-labels.csv]: to lower the repo's memory, I'm currently working on #503 

in the meantime, just keep the labels and samples local, then we will move them to some cloud :)

File: web-client/src/tasks.ts [hunk 4]
```diff
@@ -17,6 +17,8 @@ export async function loadTasks (): Promise<Set<Task>> {
   const response = await axios.get(url.href)
   const tasks: unknown = response.data
 
+  console.log(tasks)
```
Comment (s314cy) [inline web-client/src/tasks.ts]: don't forget to remove debug prints :)

File: discojs/discojs-node/src/tasks/geotags.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      loss: 'categoricalCrossentropy',
+      metrics: ['accuracy']
+    },
+    dataType: 'image',
+    IMAGE_H: 224,
+    IMAGE_W: 224,
+    preprocessingFunctions: [],
+    RESIZED_IMAGE_H: 224,
+    RESIZED_IMAGE_W: 224,
+    LABEL_LIST: ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26', ' 27'],
```
Comment (s314cy) [inline discojs/discojs-node/src/tasks/geotags.ts]: since your task is supposed to have 140 labels, I'll open an issue about providing a range instead of the full list

File: discojs/discojs-node/src/tasks/geotags.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    modelCompileData: {
+      optimizer: 'adam',
+      loss: 'categoricalCrossentropy',
+      metrics: ['accuracy']
+    },
+    dataType: 'image',
+    IMAGE_H: 224,
+    IMAGE_W: 224,
+    preprocessingFunctions: [],
+    RESIZED_IMAGE_H: 224,
```
Comment (s314cy) [inline discojs/discojs-node/src/tasks/geotags.ts]: no need to provide RESIZE_IMAGE_* if the "resize" preprocessing isn't specified

**Review by s314cy — CHANGES_REQUESTED** [review]
love the task! very original and suits the mobile env very well

File: discojs/discojs-core/src/tasks/geotags.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    validationSplit: 0.2,
+    batchSize: 10,
+    modelCompileData: {
+      optimizer: 'adam',
+      loss: 'categoricalCrossentropy',
+      metrics: ['accuracy']
+    },
+    dataType: 'image',
+    preprocessingFunctions: [ImagePreprocessing.Resize],
+    LABEL_LIST: Array.from(Array(140).keys()).map(String),
```
Comment (s314cy) [inline discojs/discojs-core/src/tasks/geotags.ts]: perfect, but maybe `immutable:Range` can help here

```suggestion
    LABEL_LIST: Range(0, 140).map(String).toArray(),
```

File: discojs/discojs-core/src/dataset/data/image_data.ts [hunk 8]
```diff
@@ -13,28 +13,32 @@ export class ImageData extends Data {
...
-    try {
-      const sample = (await dataset.take(1).toArray())[0]
-      // TODO: We suppose the presence of labels
-      // TODO: Typing (discojs-node/src/dataset/data_loader/image_loader.spec.ts)
-      if (!(typeof sample === 'object' && sample !== null)) {
-        throw new Error()
-      }
+    if (!task.trainingInformation.preprocessingFunctions?.includes(ImagePreprocessing.Resize) ||
+      !task.trainingInformation.RESIZED_IMAGE_H !== undefined ||
+      !task.trainingInformation.RESIZED_IMAGE_W !== undefined) {
```
Comment (s314cy) [inline discojs/discojs-core/src/dataset/data/image_data.ts]: maybe a bit too many negations here?

```suggestion
  if (!(
    task.trainingInformation.preprocessingFunctions?.includes(ImagePreprocessing.Resize &&
    task.trainingInformation.RESIZED_IMAGE_H !== undefined &&
    task.trainingInformation.RESIZED_IMAGE_W !== undefined
  )) {
```

**Review by s314cy — CHANGES_REQUESTED** [review]

File: discojs/discojs-core/src/dataset/data/image_data.ts (hunk 8 above)
Comment (francesco98) [reply discojs/discojs-core/src/dataset/data/image_data.ts]: yes, definitely!

File: discojs/discojs-core/src/tasks/geotags.ts (hunk 7 above)
Comment (francesco98) [reply discojs/discojs-core/src/tasks/geotags.ts:29]: Are you sure that Range exists?

**Review by s314cy — APPROVED** [review]
